FLASK/todolist_api/TaskController.py

```diff
@@ -0,0 +1,123 @@
+from flask.views import MethodView
+from flask import jsonify
+from flask_restful import abort, reqparse
+from datetime import datetime
+
+
+def parser():
+    parser_args = reqparse.RequestParser()
+
+    parser_args.add_argument('name', type=str, required=True)
+    parser_args.add_argument('assigner', type=str, required=True)
+    parser_args.add_argument('company', type=str, required=True)
+    parser_args.add_argument('deadline', type=str)
+    parser_args.add_argument('priority', type=int)
+    parser_args.add_argument('description', type=str)
+    parser_args.add_argument('status', default='In progress')
+    parser_args.add_argument('assigned_personnel', type=str)
+    parser_args.add_argument('creation_date', type=datetime,
+                             default=datetime.today())
+    parser_args.add_argument('last_modified_date', type=datetime,
+                             default=datetime.today())
+    parser_args.add_argument('comments', type=str)
+
+    args = parser_args.parse_args()
+    return args
+
+
+# Interface
+class Controller(MethodView):
+    def __init__(self, repository):
+        self.repository = repository
+
+    def get(self, task_id):
+        if task_id.lower() == "all":
+            return self.repository.get_all()
+        elif self.repository.task_exists(task_id):
+            return self.repository.get_by_id(task_id)
+        else:
+            abort(404, message=f'Task {task_id} was not found!')
+
+    def post(self):
+        if parser():
+            args = parser()
+
+            # Converting the creation date & last modified date to %Y-%m-%d format
+            creation_date = args['creation_date'].strftime('%Y-%m-%d %H:%M')
+            last_modified_date = args['last_modified_date'].strftime('%Y-%m-%d %H:%M')
+
+            # Creating a new task using the parsed arguments
+            new_data = {
```

Comment:
Aici, ai putea sa folosesti modelul de task, care are o structura si metode definite. Desi poate parea mai usor acum, s-ar putea ca partea asta sa te incurce pe viitor, deoarece:
1. Daca vrem sa schimbam unele argumente?
2. daca vrem sa fie un field de la sine inteles, mai exact name sau company, sa fie inclus automat de catre server in functie de cine foloseste aplicatia?

Daca lasam asa, ar insemna sa modificam maps in loc sa modificam obiecte si ar duce la greseli pe viitor.